### PR TITLE
Refactoring of the initialization code of the 'particleMesh' variable.

### PR DESCRIPTION
**Pull request by capdevon**
```java
switch (meshType) {
    case Point:
        particleMesh = new ParticlePointMesh();
        this.setMesh(particleMesh);
        break;
    case Triangle:
        particleMesh = new ParticleTriMesh();
        this.setMesh(particleMesh);
        break;
    default:
        throw new IllegalStateException("Unrecognized particle type: " + meshType);
}
```

This piece of code is repeated 3 times in `ParticleEmitter` class, violating the DRY programming rule. So I encapsulated it in a separate method.

Fixed typo compatability -> compatibility

### CODE REVIEW

File: jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java [hunk 1]
```diff
@@ -284,6 +284,20 @@ public ParticleEmitter(String name, Type type, int numParticles) {
...
+//        particleMesh.initParticleData(this, particles.length);
+    }
+
+    /**
+     * For serialization only. Do not use.
+     */
+    protected ParticleEmitter() {
+        super();
+        setBatchHint(BatchHint.Never);
+    }
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java:298]: Why did you move the protected constructor?

Comment (capdevon) [reply jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java:298]: I inserted the method after the default constructor. I don't know why github points out the differences so weird for this point. If you see the whole file you can check that everything is ok.

Comment (stephengold) [reply jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java:298]: GitHub's diff algorithm is unhelpful sometimes, and I guess this is one of those times. Sorry for my confusion!

**Comment by stephengold** [comment]
Thanks for your contribution to our project, @capdevon!
